content/articles/Introduction-to-Linux-Shell-and-Shell-Scripting/index.md

```diff
@@ -0,0 +1,148 @@
+
+### Table of contents
+
+- [What is the kernel](#what-is-the-kernel)
+- [What is Shell](#what-is-shell)
+- [Shell Scripting](#shell-scripting)
+- [Why do we need shell scripts](#why-do-we-need-shell-scripts)
+- [Easy demo of Bash Shell scripting](#easy-demo-of-bash-shell-scripting)
+
+### What is the kernel
+
+The `kernel` is computer software that is at the heart of a computer's operating system, controlling everything in it. It is in charge of the following Linux system's resources:
+
+- File administration. 
+- Management of processes.
+- Management of I/O.
+- Device management, memory management, and so forth. 
+
+Linus Torvalds is sometimes misidentified as the creator of the Linux operating system, but he is only responsible for the creation of the Linux kernel. 
+
+Kernel + GNU system utilities and libraries + additional management scripts + installation scripts Equals complete Linux system.
+
+### What is Shell
+
+A `shell` is a unique user program that allows users to interact with operating system functions. Shell accepts user-readable commands and converts them to kernel-friendly. It is an interpreter for commands that can read and run commands on keyboards or files. When a user logs in or launches a terminal, the shell is launched.
+
+Shells can be divided into two categories: 
+1. Command-Line shell
+2. Graphical shell
+
+#### Command-Line shell
+
+Shell access is possible through a shell interface shell. A certain tool called 'Terminal' is used to key in human-readable commands like `cat` and `ls` in Linux/macOS or 'command prompt' in Windows, which subsequently carries out tasks. The user is then shown the result on the terminal. This is how an Ubuntu  terminal looks like:
+
+![output](/engineering-education/Introduction-to-Linux-Shell-and-Shell-Scripting/ubuntu_terminal.png)
+
+The `ls` command is used with the `-l` option in the screenshot above. For a long listing, the files are displayed in the current working directory. Working with the command line shell can be challenging for beginners since there are so many commands to remember. It is extremely powerful because it allows users to save commands in a file and then execute them all at once. Any repetitious task can be easily automated this way. Batch records in Windows and Shell Scripts in Linux/macOS frameworks are normal names for these documents.
+
+#### Graphical shell
+
+The Graphical shell allow the use of a `GUI` program for open, close, move and resize windows, as well as to switch focus between window sizes. Window Operating systems or Ubuntu Operating systems are notable examples of operating systems that provide a graphical user interface (GUI) for interacting with programs. For each activity, the user does not need to enter a command. A typical Ubuntu user interface is shown below:
+
+For Linux systems, there are various shells available, such as: 
+
```

Comment:
"A typical Ubuntu user interface is shown below:"

Did you forget to include it? I can't see it